### PR TITLE
Add Gaudi CI for Sentence Transformers

### PR DESCRIPTION
**Pull request by regisss**
# What does this PR do?

As per title.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: tests/ci/sentence_transformers.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+python -m pip install --upgrade pip
+python -m pip install /root/workspace/optimum-habana[tests]
+cd /root/workspace/sentence-transformers/tests
+python -m pip install ..
+python -m pytest test_compute_embeddings.py
+cd /root/workspace/optimum-habana/tests
+python -m pytest test_sentence_transformers.py
```
Comment (libinta) [inline tests/ci/sentence_transformers.sh:9]: can we add test for all py files except training related here? https://github.com/UKPLab/sentence-transformers/tree/master/tests

File: tests/test_sentence_transformers.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import pytest
+from sentence_transformers import SentenceTransformer, util
+
+from .test_examples import TIME_PERF_FACTOR
+
+
+if os.environ.get("GAUDI2_CI", "0") == "1":
+    # Gaudi2 CI baselines
+    MODELS_TO_TEST = [
+        ("sentence-transformers/all-mpnet-base-v2", 762.5595168883357),
```
Comment (libinta) [inline tests/test_sentence_transformers.py:15]: can we have this list here ?
all-mpnet-base-v2 
multi-qa-mpnet-base-dot-v1 
all-distilroberta-v1 
all-MiniLM-L12-v2 
multi-qa-distilbert-cos-v1 
all-MiniLM-L6-v2 
multi-qa-MiniLM-L6-cos-v1 
paraphrase-multilingual-mpnet-base-v2 
paraphrase-albert-small-v2 
paraphrase-multilingual-MiniLM-L12-v2 
paraphrase-MiniLM-L3-v2 
distiluse-base-multilingual-cased-v1 
distiluse-base-multilingual-cased-v2 
E5-mistral-7b-instruct

File: tests/ci/sentence_transformers.sh (hunk 1 above)
Comment (sidgog) [reply tests/ci/sentence_transformers.sh:9]: also, this test is only for perf. can we details for accuracy test for slow test?

Comment (regisss) [reply tests/ci/sentence_transformers.sh:9]: @libinta Some of these tests fail, I'm only including the ones that pass. We'll investigate later the issues with the failing tests.

@sidgog Not sure what you mean here, accuracy is tested in the tests called at line 7 of this script.

File: tests/test_sentence_transformers.py (hunk 2 above)
Comment (regisss) [reply tests/test_sentence_transformers.py:15]: @libinta E5-mistral-7b-instruct doesn't work. Running
```py
from sentence_transformers import SentenceTransformer

model = SentenceTransformer("intfloat/e5-mistral-7b-instruct")

# Our sentences we like to encode
sentences = [
    "This framework generates embeddings for each input sentence",
    "Sentences are passed as a list of strings.",
    "The quick brown fox jumps over the lazy dog.",
]

# Sentences are encoded by calling model.encode()
embeddings = model.encode(sentences)
```
returns
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.10/dist-packages/sentence_transformers/SentenceTransformer.py", line 371, in encode
    out_features = self.forward(features)
  File "/usr/local/lib/python3.10/dist-packages/habana_frameworks/torch/hpu/graphs.py", line 661, in forward
    return wrapped_hpugraph_forward(cache, stream, orig_fwd, args, kwargs, disable_tensor_cache, asynchronous, dry_run, max_graphs)
  File "/usr/local/lib/python3.10/dist-packages/habana_frameworks/torch/hpu/graphs.py", line 545, in wrapped_hpugraph_forward
    graph.capture_end()
  File "/usr/local/lib/python3.10/dist-packages/habana_frameworks/torch/hpu/graphs.py", line 41, in capture_end
    _hpu_C.capture_end(self.hpu_graph)
RuntimeError: Graph compile failed. synStatus=synStatus 26 [Generic failure].
```

**Pull request merged**